src/aleph/vm/pool.py

```diff
@@ -188,6 +199,46 @@ def forget_vm(self, vm_hash: ItemHash) -> None:
         except KeyError:
             pass
 
+    async def _load_persistent_executions(self):
+        """Load persistent executions from the database."""
+        saved_executions = await get_execution_records()
+        for saved_execution in saved_executions:
```

Comment:
In this use case, would it make sense to use an [AsyncSession](https://docs.sqlalchemy.org/en/20/orm/extensions/asyncio.html#sqlalchemy.ext.asyncio.AsyncSession) instead of a synchronous one ?

The original implementation of `get_execution_records()` was synchronous because all data was always returned at once to the HTTP client, but that is not the case anymore with this.

Reply:
Why to complicate it? Which are the advantages of doing it with AsyncSession?

Reply:
> "When using the AsyncSession object from SQLAlchemy’s asyncio extension, this object is only a thin proxy on top of a Session, and the same rules apply; it is an unsynchronized, mutable, stateful object, so it is not safe to use a single instance of AsyncSession in multiple asyncio tasks at once."

One `Session` per thread, one `AsyncSession` per task is what is intended. An AsyncSession in an `async` function [is exactly what we want](https://docs.sqlalchemy.org/en/20/orm/session_basics.html#is-the-session-thread-safe-is-asyncsession-safe-to-share-in-concurrent-tasks).

Reply:
Also, consider a Session to be a transaction for the DB. If you have mutliple Session objects across multiple async tasks, this will lead to unnecessary blocking running python code, while the database is executing a transaction. Using AsyncSession, this will be handled gracefully.

Reply:
> Why to complicate it? Which are the advantages of doing it with AsyncSession?

Using the Session object makes database calls blocking on the main thread of the application. This means that all tasks running on the orchestrator (proxying HTTP requests, starting new VMs, ...) will be waiting for that database call to return.  Using an AsyncSession would make this wait in the background and allow the main event loop to handle other events in the meantime.

Reply:
Okay, this is solved in https://github.com/aleph-im/aleph-vm/pull/512 PR